chpl/chpl-service/src/main/java/gov/healthit/chpl/changerequest/validation/ChangeRequestCreateValidation.java

```diff
@@ -1,35 +1,30 @@
-
-import org.apache.commons.lang3.StringUtils;
-
+import gov.healthit.chpl.changerequest.domain.ChangeRequestDeveloperDemographics;
 import gov.healthit.chpl.manager.rules.ValidationRule;
 
 public class ChangeRequestCreateValidation extends ValidationRule<ChangeRequestValidationContext> {
 
     @Override
     public boolean isValid(ChangeRequestValidationContext context) {
-        if (context.getNewChangeRequest().getChangeRequestType().getId().equals(context.getChangeRequestTypeIds().getWebsiteChangeRequestType())
-                && (context.getNewChangeRequest().getDetails() == null
-                        || !isChangeRequestWebsiteValid((HashMap) context.getNewChangeRequest().getDetails()))) {
-            getMessages().add(getErrorMessage("changeRequest.details.website.invalid"));
-            return false;
-        } else if (context.getNewChangeRequest().getChangeRequestType().getId().equals(context.getChangeRequestTypeIds().getDeveloperDetailsChangeRequestTypeId())) {
+        if (context.getNewChangeRequest().getChangeRequestType().getId().equals(context.getChangeRequestTypeIds().getDeveloperDemographicsChangeRequestTypeId())) {
+            ChangeRequestDeveloperDemographics details = (ChangeRequestDeveloperDemographics) context.getNewChangeRequest().getDetails();
             if (context.getNewChangeRequest().getDetails() == null) {
-                getMessages().add(getErrorMessage("changeRequest.details.invalid"));
+                getMessages().add(getErrorMessage("changeRequest.demographics.invalid"));
                 return false;
             } else {
                 boolean areDetailsValid = true;
-                HashMap<String, Object> crDetails = (HashMap) context.getNewChangeRequest().getDetails();
-                if (!isChangeRequestSelfDevloperValid(crDetails)) {
-                    getMessages().add(getErrorMessage("changeRequest.details.selfDeveloper.missing"));
+                if (!isChangeRequestSelfDevloperValid(details)) {
+                    getMessages().add(getErrorMessage("changeRequest.demographics.selfDeveloper.missing"));
+                    areDetailsValid = false;
+                } else if (!isChangeRequestAddressValid(details)) {
+                    getMessages().add(getErrorMessage("changeRequest.demographics.address.missing"));
                     areDetailsValid = false;
-                } else if (!isChangeRequestAddressValid(crDetails)) {
-                    getMessages().add(getErrorMessage("changeRequest.details.address.missing"));
+                } else if (!isChangeRequestContactValid(details)) {
+                    getMessages().add(getErrorMessage("changeRequest.demographics.contact.missing"));
                     areDetailsValid = false;
-                } else if (!isChangeRequestContactValid(crDetails)) {
-                    getMessages().add(getErrorMessage("changeRequest.details.contact.missing"));
+                } else if (!isChangeRequestContactValid(details)) {
```

Comment:
It looks like this "else if" is the same as the one above it. Am I reading this diff right?

Reply:
Good catch - it was a duplicate.